### PR TITLE
add detect custom tsconfig env var

### PR DESCRIPTION
**Pull request by lillianzhang331**
# Description

This is for supporting the configuration of the default tsconfig file. It will use an environment variable to detect when a custom file is used (such as tsconfig.server.json). 

# Checklist:

- [x] Run these changes with `pack`
- [x] Make updates to `CHANGELOG.md`

### CODE REVIEW

File: bin/build [hunk 1]
```diff
@@ -12,3 +12,10 @@ source "$bp_dir/lib/build.sh"
 if check_tsc_binary "$build_dir" ; then
   export PATH=./node_modules/typescript/bin:$PATH
 fi
+
+# shellcheck disable=SC2034
+if detect_tsconfig_env_var ; then
+  tsconfig="$CUSTOM_TSCONFIG"
+else
+  tsconfig="tsconfig.json"
+fi
```
Comment (danielleadams) [inline bin/build]: I think we can just make this something like:
```sh
TSCONFIG=${CUSTOM_TSCONFIG:-"tsconfig.json"}
```
(https://stackoverflow.com/a/12691027/4135520)

Comment (lillianzhang331) [reply bin/build]: wow so nifty, that simplified everything lol

File: bin/build [hunk 2]
```diff
@@ -12,3 +12,6 @@ source "$bp_dir/lib/build.sh"
 if check_tsc_binary "$build_dir" ; then
   export PATH=./node_modules/typescript/bin:$PATH
 fi
+
+# shellcheck disable=SC2034
+TSCONFIG=${CUSTOM_TSCONFIG:-"tsconfig.json"}
```
Comment (danielleadams) [inline bin/build:17]: Does it still need the shellcheck disable?

**Comment by danielleadams** [comment]
Hey @lillianzhang331 I think there was a test with this that disappeared 😄

File: bin/build (hunk 2 above)
Comment (lillianzhang331) [reply bin/build:17]: Yeah, I think so. Until we use the tsconfig variable, shellcheck is going to mark it as unused.

**Comment by lillianzhang331** [comment]
@danielleadams I deleted the test because I was no longer using a method to determine what the tsconfig variable so I assumed it would make more sense to create the test once the variable is used instead. Let me know if I'm wrong to assume that and I can add a test in. Can probably have it echo the variable once it is set for testing purposes?

**Comment by danielleadams** [comment]
Gotcha - I think that's fine then. If we test with the second part of this, I'm good with that.

**Review by danielleadams — APPROVED** [review]

**Pull request merged**